test/testcafe/spec/DeviceChallengePollView_spec.js

```diff
@@ -247,13 +247,19 @@ const universalLinkMock = RequestMock()
 const appLinkWithoutLaunchMock = RequestMock()
   .onRequestTo(/idp\/idx\/introspect/)
   .respond(identifyWithDeviceProbingLoopback)
+  .onRequestTo(request => {
+    return request.url.match(/\/idp\/idx\/authenticators\/poll$/) && !appLinkLoopBackFailed;
+  })
+  .respond(identifyWithDeviceProbingLoopback)
   .onRequestTo(/(2000|6511|6512|6513)\/probe/)
   .respond(null, 500, { 'access-control-allow-origin': '*' })
   .onRequestTo(/\/idp\/idx\/authenticators\/poll\/cancel/)
   .respond(identifyWithLoopbackFallbackAndroidWithoutLink)
   .onRequestTo(/\/idp\/idx\/authenticators\/okta-verify\/launch/)
   .respond(identifyWithLaunchAppLink)
-  .onRequestTo(/\/idp\/idx\/authenticators\/poll/)
+  .onRequestTo(request => {
+    return request.url.match(/\/idp\/idx\/authenticators\/poll/) && appLinkLoopBackFailed;
+  })
```

Comment:
nit: are you able to place this up with the other /poll/ request? Also is the lack of the `$` at the end intentional?